src/responses/search-response-details.ts

```diff
@@ -249,6 +295,51 @@ export class SearchResponseDetails implements SearchResponseDetailsInterface {
     return hits;
   }
 
+  /**
+   * Special handling for when the federated search elements are present in the response.
+   * Formats all non-metadata hits to a new federatedResults object in the results.
+   */
+  private handleFederatedPageElements(): void {
+    const SEARCH_SERVICES: FederatedServiceName[] = [
+      'service___fts',
+      'service___tvs',
+      'service___rcs',
+      'service___whisper',
+    ];
+
+    for (const service of SEARCH_SERVICES) {
+      const simpleServiceName: string = this.removeServicePrefix(service);
+
+      // Add service name to federated results section
+      this.federatedResults
+        ? (this.federatedResults[service] = [])
+        : (this.federatedResults = { [simpleServiceName]: [] });
```

Comment:
SUGGESTION: Ah, you're using the ternary to execute code instead of for assignment. Interesting! While valid, I don't think it's the typical/expected use of the ternary, though I had to go online to verify this. Thus, I'd probably prefer a more boring if-else here (and anywhere else this usage occurs).

At the same time, I think it'd be a interesting discussion point to bring up in our team chat to see what others think.

Reply:
I'm down for an if/else! What we lose in ~ elegance ~ we gain in readability. 🙂